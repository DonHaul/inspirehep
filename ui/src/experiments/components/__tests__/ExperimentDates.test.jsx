import React from 'react';
import { render } from '@testing-library/react';

import ExperimentDates from '../ExperimentDates';

describe('ExperimentDates', () => {
  it('renders with all props set', () => {
    const { asFragment } = render(
      <ExperimentDates
        dateApproved="1984-02-02"
        dateProposed="1984-02-01"
        dateStarted="1984-02-03"
        dateCancelled="1984-02-04"
        dateCompleted="1984-02-05"
        wrapperClassName="di"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders without dateCancelled or dateCompleted', () => {
    const { asFragment } = render(
      <ExperimentDates
        dateApproved="1984-02-02"
        dateProposed="1984-02-01"
        dateStarted="1984-02-03"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with partial dates', () => {
    const { asFragment } = render(
      <ExperimentDates
        dateApproved="1984-02"
        dateProposed="1984"
        dateStarted="1984-02-03"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
