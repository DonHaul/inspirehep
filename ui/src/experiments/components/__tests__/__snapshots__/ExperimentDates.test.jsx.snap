// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentDates renders with all props set 1`] = `
<DocumentFragment>
  <div
    class="__InlineList__ di"
  >
    <ul>
      <li>
        <span>
          Proposed: Feb 1, 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Approved: Feb 2, 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Started: Feb 3, 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Cancelled: Feb 4, 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Completed: Feb 5, 1984
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`ExperimentDates renders with partial dates 1`] = `
<DocumentFragment>
  <div
    class="__InlineList__"
  >
    <ul>
      <li>
        <span>
          Proposed: 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Approved: Feb, 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Started: Feb 3, 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Still Running
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`ExperimentDates renders without dateCancelled or dateCompleted 1`] = `
<DocumentFragment>
  <div
    class="__InlineList__"
  >
    <ul>
      <li>
        <span>
          Proposed: Feb 1, 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Approved: Feb 2, 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Started: Feb 3, 1984
        </span>
        , 
      </li>
      <li>
        <span>
          Still Running
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
